quartz/plugins/transformers/roam.ts

```diff
@@ -0,0 +1,132 @@
+  TODOComponent: boolean
+  DONEComponent: boolean
+  videoComponent: boolean
+  audioComponent: boolean
+  pdfComponent: boolean
+  blockquoteComponent: boolean
+  tableComponent: boolean
+  attributeComponent: boolean
+}
+
+const defaultOptions: Options = {
+  orComponent: true,
+  TODOComponent: true,
+  DONEComponent: true,
+  videoComponent: true,
+  audioComponent: true,
+  pdfComponent: true,
+  blockquoteComponent: true,
+  tableComponent: true,
+  attributeComponent: true,
+}
+
+const orRegex = new RegExp(/{{or:(.*?)}}/, "g")
+const TODORegex = new RegExp(/{{.*?\bTODO\b.*?}}/, "g")
+const DONERegex = new RegExp(/{{.*?\bDONE\b.*?}}/, "g")
+const videoRegex = new RegExp(/{{.*?\bvideo\b.*?\:(.*?)}}/, "g")
+const youtubeRegex = new RegExp(/{{.*?\bvideo\b.*?(\bhttp.*?\byoutu.*?)watch\?v=(.*?)}}/, "g")
+
+const audioRegex = new RegExp(/{{.*?\baudio\b.*?\:(.*?)}}/, "g")
+const pdfRegex = new RegExp(/{{.*?\bpdf\b.*?\:(.*?)}}/, "g")
+const blockquoteRegex = new RegExp(/\[\[>\]\]/, "g")
+const roamHighlightRegex = new RegExp(/\^\^(.+)\^\^/, "g")
+const roamItalicRegex = new RegExp(/__(.+)__/, "g")
+const tableRegex = new RegExp(/- {{.*?\btable\b.*?}}/, "g") /* TODO */
+const attributeRegex = new RegExp(/\b\w+(?:\s+\w+)*::/, "g") /* TODO */
+
+export const RoamFlavoredMarkdown: QuartzTransformerPlugin<Partial<Options> | undefined> = (
+  userOpts,
+) => {
+  const opts = { ...defaultOptions, ...userOpts }
+
+  return {
+    name: "RoamFlavoredMarkdown",
+    textTransform(_ctx, src) {
+      if (opts.orComponent) {
+        src = src.toString()
+        src = src.replaceAll(orRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const options = match.split("|")
+          const id = Math.random().toString(36).substring(2, 15) // Generate a unique ID
+          const dropdown = `<select class="roam-or-component">${options.map((option: string) => `<option>${option}</option>`).join("")}</select>`
+          return dropdown
+        })
+      }
+      if (opts.TODOComponent) {
+        src = src.toString()
+        src = src.replaceAll(TODORegex, (value, ...capture) => {
+          const checkbox = '<input type="checkbox" disabled>'
+          return checkbox
+        })
+      }
+      if (opts.DONEComponent) {
+        src = src.toString()
+        src = src.replaceAll(DONERegex, (value, ...capture) => {
+          const checkbox = '<input type="checkbox" checked disabled>'
+          return checkbox
+        })
+      }
+      if (opts.videoComponent) {
+        //youtube first then regular video links
+        src = src.toString()
+        src = src.replaceAll(youtubeRegex, (value, ...capture) => {
+          const [match, text] = capture
+          const video = `<iframe width="560" height="315" src="https://www.youtube.com/embed/${text}" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>`
```

Comment:
Can you check if this can produce the same output as the YouTube embed in the Obsidian Flavored Markdown plugin? (width, height, options, etc.)